homeassistant/components/cloud/http_api.py

```diff
@@ -81,39 +90,133 @@ class CloudLogoutView(HomeAssistantView):
     name = 'api:cloud:account'
 
     @asyncio.coroutine
     def get(self, request):
-        """Validate config and return results."""
+        """Get account info."""
         hass = request.app['hass']
+        auth = hass.data['cloud']['auth']
 
-        if 'cloud' not in hass.data[DOMAIN]:
+        if not auth.is_logged_in:
             return self.json_message('Not logged in', 400)
 
-        return self.json(hass.data[DOMAIN]['cloud'].account)
+        return self.json(_auth_data(auth))
+
+
+class CloudRegisterView(HomeAssistantView):
+    """Register on the Home Assistant cloud."""
+
+    url = '/api/cloud/register'
+    name = 'api:cloud:register'
+
+    @asyncio.coroutine
+    @_handle_cloud_errors
+    @RequestDataValidator(vol.Schema({
+        vol.Required('email'): str,
+        vol.Required('password'): vol.All(str, vol.Length(min=6)),
+    }))
+    def post(self, request, data):
+        """Handle registration request."""
+        hass = request.app['hass']
+
+        with async_timeout.timeout(REQUEST_TIMEOUT, loop=hass.loop):
+            yield from hass.async_add_job(
+                auth_api.register, hass, data['email'], data['password'])
+
+        return self.json_message('ok')
+
+
+class CloudConfirmRegisterView(HomeAssistantView):
+    """Confirm registration on the Home Assistant cloud."""
+
+    url = '/api/cloud/confirm_register'
+    name = 'api:cloud:confirm_register'
+
+    @asyncio.coroutine
+    @_handle_cloud_errors
+    @RequestDataValidator(vol.Schema({
+        vol.Required('confirmation_code'): str,
+        vol.Required('email'): str,
+    }))
+    def post(self, request, data):
+        """Handle registration confirmation request."""
+        hass = request.app['hass']
+
+        with async_timeout.timeout(REQUEST_TIMEOUT, loop=hass.loop):
+            yield from hass.async_add_job(
+                auth_api.confirm_register, hass, data['confirmation_code'],
+                data['email'])
+
+        return self.json_message('ok')
+
+
+class CloudForgotPasswordView(HomeAssistantView):
+    """View to start Forgot Password flow.."""
+
+    url = '/api/cloud/forgot_password'
+    name = 'api:cloud:forgot_password'
+
+    @asyncio.coroutine
+    @_handle_cloud_errors
+    @RequestDataValidator(vol.Schema({
+        vol.Required('email'): str,
+    }))
+    def post(self, request, data):
+        """Handle forgot password request."""
+        hass = request.app['hass']
+
+        with async_timeout.timeout(REQUEST_TIMEOUT, loop=hass.loop):
+            yield from hass.async_add_job(
+                auth_api.forgot_password, hass, data['email'])
+
+        return self.json_message('ok')
+
+
+class CloudConfirmForgotPasswordView(HomeAssistantView):
+    """View to finish Forgot Password flow.."""
+
+    url = '/api/cloud/confirm_forgot_password'
+    name = 'api:cloud:confirm_forgot_password'
+
+    @asyncio.coroutine
+    @_handle_cloud_errors
+    @RequestDataValidator(vol.Schema({
```

Comment:
I think we should defined the Schema not inline. Maybe in `cloud/const.py` or `cloud/schema.py`